e2e/src/editCra.e2e-spec.ts

```diff
@@ -0,0 +1,187 @@
+      clientInput: ElementFinder,
+      selectedNumber: ElementFinder;
+
+  const getBusinessDaysNumber = function(): number {
+    let workingDays = 0;
+    const date = startOfMonth(new Date());
+
+    for (let day = 1; day <= getDate(lastDayOfMonth(date)); day++) {
+      const weekDay = getDay(setDate(date, day));
+      workingDays += (weekDay === 0 || weekDay === 6) ? 0 : 1;
+    }
+
+    return workingDays;
+  };
+
+  beforeEach(() => {
+    editCraPage = new EditCraPage();
+    editCraPage.navigateTo();
+    browser.waitForAngular();
+
+    nameInput = editCraPage.getInputByLabelText('Nom');
+    emailInput = editCraPage.getInputByLabelText('Email');
+    titleInput = editCraPage.getInputByLabelText('Intitulé');
+    clientInput = editCraPage.getInputByLabelText('Client final');
+
+    selectedNumber = editCraPage.getElementsByText('Nombre total de journées :')
+            .first()
+            .element(by.xpath('..//span[@class="highlight"]'));
+  });
+
+  it('should be possible to enter my name', () => {
+    nameInput.sendKeys('Tester');
+    expect(nameInput.getAttribute('value')).toBe('Tester');
+  });
+
+  it('should be possible to enter my email', () => {
+    emailInput.sendKeys('tester@test.com');
+    expect(emailInput.getAttribute('value')).toBe('tester@test.com');
+  });
+
+  it('should be possible to enter a mission title', () => {
+    titleInput.sendKeys('Testing');
+    expect(titleInput.getAttribute('value')).toBe('Testing');
+  });
+
+  it('should be possible to enter the client\'s name', () => {
+    clientInput.sendKeys('Test.com');
+    expect(clientInput.getAttribute('value')).toBe('Test.com');
+  });
+
+  describe('On the calendar', () => {
+    let aDayCell: ElementFinder;
+
+    beforeEach(() => {
+      aDayCell = editCraPage.getElementsByText('4').first();
+      aDayCell.click();
+    });
+
+    it('should be possible to select days', () => {
+      expect(editCraPage.getADayBadge(aDayCell).getText()).toBe('1\njour');
+    });
+
+    it('should be possible to select every working day', () => {
+      editCraPage.getElementsByText('Sélectionner les jours ouvrés').first().click();
+      expect(selectedNumber.getText()).toBe(getBusinessDaysNumber().toString());
+    });
+
+    describe('', () => {
+      let dayBadge: ElementFinder;
+
+      beforeEach(() => {
+        dayBadge = editCraPage.getADayBadge(aDayCell);
+
+        browser.actions()
+            .mouseMove(dayBadge)
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(50);
+
+        browser.actions()
+            .click(editCraPage.getElementsByText('0,5').first())
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(50);
+
+        // Moving the mouse away to read the value
+        browser.actions()
+            .mouseMove(aDayCell)
+            .perform();
+
+        // Sleeping to wait for the annimation end
+        browser.sleep(170);
```

Comment:
The e2e are always very long, especially because of the many sleep we write in the code.
Try to reduce them to the maximum, and use something else if it's possible.
Some people disable css animation on protractor, I don't know if it's a good practice